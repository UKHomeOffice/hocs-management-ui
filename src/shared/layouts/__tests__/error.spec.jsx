import React from 'react';
import Error from '../error.tsx';
import { render } from '@testing-library/react';


describe('Error component', () => {
    it('should render with default props', () => {
        expect(
            render(<Error />)
        ).toMatchSnapshot();
    });
    it('should render with location when passed', () => {
        const props = {
            error: {
                location: {
                    pathname: '/some/unknown/path'
                }
            }
        };
        expect(
            render(<Error {...props} />)
        ).toMatchSnapshot();
    });
    it('should render with location when 404 passed', () => {
        const props = {
            error: {
                status: 404,
                location: {
                    pathname: '/SOME/UNSUPPORTED/ENDPOINT'
                }
            }
        };
        expect(
            render(<Error {...props} />)
        ).toMatchSnapshot();
    });
    it('should render with location when 403 passed', () => {
        const props = {
            error: {
                status: 403,
                location: {
                    pathname: '/SOME/UNSUPPORTED/ENDPOINT'
                }
            }
        };
        expect(
            render(<Error {...props} />)
        ).toMatchSnapshot();
    });
    it('should render with a stack trace when passed', () => {
        const props = {
            error: {
                stack: 'Stack trace...'
            }
        };
        expect(
            render(<Error {...props} />)
        ).toMatchSnapshot();
    });
});
