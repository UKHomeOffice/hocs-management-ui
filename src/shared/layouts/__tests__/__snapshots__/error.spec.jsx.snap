// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error component should render with a stack trace when passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-full"
        >
          <h1
            class="govuk-heading-xl"
          >
            Something has gone wrong
          </h1>
          <p
            class="govuk-body"
          >
            Please try again
          </p>
          <details
            class="govuk-details"
            open=""
          >
            <summary
              class="govuk-details__summary"
            >
              <span
                class="govuk-details__summary-text"
              >
                Stack trace
              </span>
            </summary>
            <div
              class="govuk-details__text"
            >
              <pre
                class="code overflow-scroll"
              >
                Stack trace...
              </pre>
            </div>
          </details>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="govuk-grid-row"
    >
      <div
        class="govuk-grid-column-full"
      >
        <h1
          class="govuk-heading-xl"
        >
          Something has gone wrong
        </h1>
        <p
          class="govuk-body"
        >
          Please try again
        </p>
        <details
          class="govuk-details"
          open=""
        >
          <summary
            class="govuk-details__summary"
          >
            <span
              class="govuk-details__summary-text"
            >
              Stack trace
            </span>
          </summary>
          <div
            class="govuk-details__text"
          >
            <pre
              class="code overflow-scroll"
            >
              Stack trace...
            </pre>
          </div>
        </details>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Error component should render with default props 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-full"
        >
          <h1
            class="govuk-heading-xl"
          >
            Something has gone wrong
            
          </h1>
          <p
            class="govuk-body"
          >
            Please try again
          </p>
          
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="govuk-grid-row"
    >
      <div
        class="govuk-grid-column-full"
      >
        <h1
          class="govuk-heading-xl"
        >
          Something has gone wrong
          
        </h1>
        <p
          class="govuk-body"
        >
          Please try again
        </p>
        
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Error component should render with location when 403 passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-full"
        >
          <h1
            class="govuk-heading-xl"
          >
            You do not have permission
          </h1>
          <p
            class="govuk-body"
          >
            If you expect permission to perform this action, raise a ticket.
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="govuk-grid-row"
    >
      <div
        class="govuk-grid-column-full"
      >
        <h1
          class="govuk-heading-xl"
        >
          You do not have permission
        </h1>
        <p
          class="govuk-body"
        >
          If you expect permission to perform this action, raise a ticket.
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Error component should render with location when 404 passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-full"
        >
          <h1
            class="govuk-heading-xl"
          >
            Page does not exist
          </h1>
          <p
            class="govuk-body"
          >
            If you typed the web address, check it is correct.
          </p>
          <p
            class="govuk-body"
          >
            If you pasted the web address, check you copied the entire address.
          </p>
          <p>
            <code
              class="code"
            >
              /SOME/UNSUPPORTED/ENDPOINT
            </code>
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="govuk-grid-row"
    >
      <div
        class="govuk-grid-column-full"
      >
        <h1
          class="govuk-heading-xl"
        >
          Page does not exist
        </h1>
        <p
          class="govuk-body"
        >
          If you typed the web address, check it is correct.
        </p>
        <p
          class="govuk-body"
        >
          If you pasted the web address, check you copied the entire address.
        </p>
        <p>
          <code
            class="code"
          >
            /SOME/UNSUPPORTED/ENDPOINT
          </code>
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Error component should render with location when passed 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="govuk-grid-row"
      >
        <div
          class="govuk-grid-column-full"
        >
          <h1
            class="govuk-heading-xl"
          >
            Something has gone wrong
          </h1>
          <p
            class="govuk-body"
          >
            Please try again
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="govuk-grid-row"
    >
      <div
        class="govuk-grid-column-full"
      >
        <h1
          class="govuk-heading-xl"
        >
          Something has gone wrong
        </h1>
        <p
          class="govuk-body"
        >
          Please try again
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
